addons/mail/models/discuss/res_partner.py

```diff
@@ -81,25 +76,20 @@ def get_mention_suggestions_from_channel(self, channel_id, search, limit=8):
         channel = self.env["discuss.channel"].search([("id", "=", channel_id)])
         if not channel:
             return []
-        domain = expression.AND(
+        domain = Domain.AND(
```

Comment:
This one would be simpler with `&` if `_get_mention_suggestions_domain` was changed to return actual `Domain` too.